test/e2e/apimachinery/aggregator.go

```diff
@@ -47,9 +49,20 @@ import (
 var serverAggregatorVersion = utilversion.MustParseSemantic("v1.7.0")
 
 var _ = SIGDescribe("Aggregator", func() {
+	var ns string
+	var c clientset.Interface
+	var aggrclient *aggregatorclient.Clientset
 	f := framework.NewDefaultFramework("aggregator")
 	framework.AddCleanupAction(func() {
-		cleanTest(f)
+		if c != nil && aggrclient != nil {
```

Comment:
We can remove these nil checks. They shouldn't be `nil` if cleanup is called as expected

Reply:
@janetkuo  We may need the nil checks.
`f.ClientSet` isn't created by framework.NewDefaultFramework("aggregator") directly but created by `f.BeforeEach` , see:
https://github.com/kubernetes/kubernetes/blob/02b3928a921b2e0930bf67b48f71c9dcd1086b4f/test/e2e/framework/framework.go#L126-L138

if we run test case like:
```
go run hack/e2e.go -v -test --test_args='--ginkgo.focus="xxx"'
```
some test cases may be skipped, `f.BeforeEach` will not be called, but `framework.AddCleanupAction(func())` will be called. In this situation, the value of `f.ClientSet` will be `nil`.

Reply:
Ah, you're right. However it seems wrong to have skipped test add cleanup actions. We should fix that (not here, in a follow up PR). 

`f.Namespace` should be set unless it's skipped. Just checking namespace should be enough (`if len(ns) > 0 {...}`).

```go
type Framework struct {
	Namespace                *v1.Namespace   // Every test has at least one namespace unless creation is skipped
}
```

Would you also add a comment here (and everywhere else) to explain this? Something like:
```go 
// Cleanup actions will be called even when the tests are skipped, leaving namespace unset.
```